engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -573,6 +573,29 @@ describe('System test: QuerySparql', () => {
         expect((await arrayifyStream(await result.execute()))).toEqual([]);
       });
     });
+
+    // TODO: better name here before merge.
+    describe('Comunica sparql regression test for #1029', () => {
+      it('Supports EXISTS sparql keyword', async() => {
+        const s = DF.namedNode('s');
+        const p = DF.namedNode('p');
+        const o = DF.namedNode('o');
+        const s1 = DF.namedNode('s1');
+
+        const nodes = [ DF.quad(s, p, o), DF.quad(s1, p, o) ];
+
+        const store = new Store(nodes);
+
+        const result = await engine.queryQuads(
+          `CONSTRUCT {?s ?p ?o.} WHERE {?s ?p ?o. OPTIONAL {FILTER EXISTS {?s2 ?p ?o. FILTER (?s != ?s2)}}}`,
+          { sources: [ store ]},
+        );
+        expect(await arrayifyStream(result)).toEqual([
```

Comment:
```suggestion
        expect(await arrayifyStream(result)).toEqualRdfQuadArray([
```

Changing the equality checker should resolve the problem.

More details: https://github.com/rubensworks/jest-rdf